quarkus-hibernate-orm extension is breaking native-image generation

Java version: OpenJDK GraalVM CE 1.0.0-rc16
OS: MacOS (but also breaks the docker image generation)
Quarkus Version : <quarkus.version>0.15.0</quarkus.version>


```
<dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-hibernate-orm</artifactId>
</dependency>
```

`mvn package -Pnative`

is throwing the following errors:

```
[INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] Running Quarkus native-image plugin on OpenJDK GraalVM CE 1.0.0-rc16
[INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] /Library/Java/JavaVirtualMachines/graalvm-ce-1.0.0-rc16/Contents/Home/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dio.netty.leakDetection.level=DISABLED -J-Dvertx.disableDnsResolver=true -J-Dio.netty.noUnsafe=true -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar profiler-1.0-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+PrintAnalysisCallTree -H:-AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services -H:-SpawnIsolates -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace
[profiler-1.0-SNAPSHOT-runner:57017]    classlist:  10,221.40 ms
[profiler-1.0-SNAPSHOT-runner:57017]        (cap):   1,516.96 ms
[profiler-1.0-SNAPSHOT-runner:57017]        setup:   3,081.53 ms
08:37:15,654 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Beta3
08:37:15,944 INFO  [com.arj.ats.arjuna] ARJUNA012170: TransactionStatusManager started on port 52432 and host 127.0.0.1 with service com.arjuna.ats.arjuna.recovery.ActionStatusService
08:37:16,044 INFO  [org.xnio] XNIO version 3.7.0.Final
08:37:16,098 INFO  [org.xni.nio] XNIO NIO Implementation Version 3.7.0.Final
08:37:39,287 INFO  [org.hib.Version] HHH000412: Hibernate Core {5.4.2.Final}
[profiler-1.0-SNAPSHOT-runner:57017]     analysis:  46,709.21 ms
Printing call tree to /Users/foo/Development/Java/application/target/reports/call_tree_profiler-1.0-SNAPSHOT-runner_2019 0517_083812.txt
Printing list of used classes to /Users/foo/Development/Java/application/target/reports/used_classes_profiler-1.0-SNAPSHOT-runner_20190517_083815.txt
Printing list of used packages to /Users/foo/Development/Java/application/target/reports/used_packages_profiler-1.0-SNAPSHOT-runner_20190517_083815.txt
Error: Unsupported features in 3 methods
Detailed message:
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getConstantPool(System.java:1227)
Call path from entry point to java.lang.System$2.getConstantPool(Class): 
	at java.lang.System$2.getConstantPool(System.java:1227)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:323)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getConstantPool(java.lang.Class):s
	java.lang.System$2.getConstantPool(java.lang.Class):s
	sun.misc.JavaLangAccess.getConstantPool(java.lang.Class):s
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getExecutableTypeAnnotationBytes(Executable) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
Call path from entry point to java.lang.System$2.getRawExecutableTypeAnnotations(Executable): 
	at java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:318)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.misc.JavaLangAccess.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getRawTypeAnnotations() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
Call path from entry point to java.lang.System$2.getRawClassTypeAnnotations(Class): 
	at java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:315)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	sun.misc.JavaLangAccess.getRawClassTypeAnnotations(java.lang.Class):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036




com.oracle.svm.core.util.UserError$UserException: Unsupported features in 3 methods
Detailed message:
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getConstantPool(System.java:1227)
Call path from entry point to java.lang.System$2.getConstantPool(Class): 
	at java.lang.System$2.getConstantPool(System.java:1227)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:323)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getConstantPool(java.lang.Class):s
	java.lang.System$2.getConstantPool(java.lang.Class):s
	sun.misc.JavaLangAccess.getConstantPool(java.lang.Class):s
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getExecutableTypeAnnotationBytes(Executable) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
Call path from entry point to java.lang.System$2.getRawExecutableTypeAnnotations(Executable): 
	at java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:318)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.misc.JavaLangAccess.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getRawTypeAnnotations() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
Call path from entry point to java.lang.System$2.getRawClassTypeAnnotations(Class): 
	at java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:315)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	sun.misc.JavaLangAccess.getRawClassTypeAnnotations(java.lang.Class):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036




	at com.oracle.svm.core.util.UserError.abort(UserError.java:75)
	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:218)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:734)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:524)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:442)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 3 methods
Detailed message:
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getConstantPool(System.java:1227)
Call path from entry point to java.lang.System$2.getConstantPool(Class): 
	at java.lang.System$2.getConstantPool(System.java:1227)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:323)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getConstantPool(java.lang.Class):s
	java.lang.System$2.getConstantPool(java.lang.Class):s
	sun.misc.JavaLangAccess.getConstantPool(java.lang.Class):s
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getExecutableTypeAnnotationBytes(Executable) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
Call path from entry point to java.lang.System$2.getRawExecutableTypeAnnotations(Executable): 
	at java.lang.System$2.getRawExecutableTypeAnnotations(System.java:1245)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:318)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	java.lang.System$2.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.misc.JavaLangAccess.getRawExecutableTypeAnnotations(java.lang.reflect.Executable):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036



Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getRawTypeAnnotations() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Trace: 
	at parsing java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
Call path from entry point to java.lang.System$2.getRawClassTypeAnnotations(Class): 
	at java.lang.System$2.getRawClassTypeAnnotations(System.java:1242)
	at sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(TypeAnnotationParser.java:315)
	at sun.reflect.annotation.TypeAnnotationParser.fetchBounds(TypeAnnotationParser.java:299)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:244)
	at sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(TypeAnnotationParser.java:237)
	at sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds(TypeVariableImpl.java:241)
	at com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036_703.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.undertow.protocols.alpn.OpenSSLAlpnProvider.setProtocols(OpenSSLAlpnProvider.java:70)
	at io.undertow.server.protocol.http.AlpnOpenListener$1$1.run(AlpnOpenListener.java:296)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeHooks(RuntimeSupport.java:144)
	at com.oracle.svm.core.jdk.RuntimeSupport.executeStartupHooks(RuntimeSupport.java:89)
	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:145)
	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
--------------------------------------------------------------------------------------------
-- WARNING: The above stack trace is not a real stack trace, it is a theoretical call tree---
-- If an interface has multiple implementations SVM will just display one potential call  ---
-- path to the interface. This is often meaningless, and what you actually need to know is---
-- the path to the constructor of the object that implements this interface.              ---
-- Quarkus has attempted to generate a more meaningful call flow analysis below          ---
---------------------------------------------------------------------------------------------

Possible path to java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	java.lang.System$2.getRawClassTypeAnnotations(java.lang.Class):b
	sun.misc.JavaLangAccess.getRawClassTypeAnnotations(java.lang.Class):b
	sun.reflect.annotation.TypeAnnotationParser.parseAllTypeAnnotations(java.lang.reflect.AnnotatedElement):s
	sun.reflect.annotation.TypeAnnotationParser.fetchBounds(java.lang.reflect.GenericDeclaration):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int, sun.reflect.annotation.TypeAnnotation$LocationInfo):j
	sun.reflect.annotation.TypeAnnotationParser.parseAnnotatedBounds(java.lang.reflect.Type[], java.lang.reflect.GenericDeclaration, int):j
	sun.reflect.generics.reflectiveObjects.TypeVariableImpl.getAnnotatedBounds():j
	java.lang.reflect.TypeVariable.getAnnotatedBounds():j
	com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036.invoke(java.lang.Object, java.lang.Object[]):j
This is an implementation of sun.reflect.MethodAccessor printing path to constructors of com.oracle.svm.reflect.TypeVariable_getAnnotatedBounds_88ea462da2b17ea45028db307519aaeeec9a4036




	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:133)
	at com.oracle.graal.pointsto.BigBang.finish(BigBang.java:564)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:685)
	... 7 more
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:28 min
[INFO] Finished at: 2019-05-17T08:38:20+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:0.15.0:native-image (default) on project profiler: Failed to generate a native image: Failed to build native image: Image generation failed -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```


if I comment the dependency in pom.xml , everything starts working as expected

So debugging this a bit, I have been able to narrow this down and solve one part of this issue. I can reproduce the issue consistently if I use a simple application with a dependency on the `quarkus-hibernate-orm` extension (and no other extension) and then run:

```
mvn clean verify -Pnative
```

(even against latest upstream master). With that command it throws a bunch of substrate exceptions which are hard to understand. So I introduced the 

```
<additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
```

in my pom.xml for the native image build and that showed up some interesting exceptions. For example of them is:

```
Fatal error: java.lang.NoClassDefFoundError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:461)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:310)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:448)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:113)
Caused by: java.lang.NoClassDefFoundError: javax/security/jacc/PolicyContextException
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.getDeclaredMethods(Class.java:1975)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleDeletedClass(AnnotationSubstitutionProcessor.java:437)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:270)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:230)
	at com.oracle.svm.hosted.NativeImageGenerator.createDeclarativeSubstitutionProcessor(NativeImageGenerator.java:875)
	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:824)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:524)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:444)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: javax.security.jacc.PolicyContextException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 15 more

```

Debugging this showed that, this exception happens because the Hibernate ORM extension has this `@Delete` substitution:


```
@TargetClass(className = "org.hibernate.secure.internal.StandardJaccServiceImpl")
@Delete
public final class Delete_StandardJaccServiceImpl {
}
```

(I'm not an exepert, but) the way Substrate seems to deal with deletes is that it loads the target class (`org.hibernate.secure.internal.StandardJaccServiceImpl`) first and then does a `getDeclaredMethods` and `getDeclaredFields` on it, which obviously means that it needs the necessary dependency classes to be present (or else you will run into the exception above). `StandardJaccServiceImpl` has depedency on jacc classes. Looking at the `quarkus-hiberate-orm` extension it doesn't list the dependency on the artifact which brings in these `javax.security.jacc.*` classes. Looking at the upstream `hibernate-orm` project, that dependency is actually a `provided` scope dependency[1], so the inclusion of `hibernate-core` artifact doesn't bring this in. So I updated our `quarkus-hibernate-orm` extension to bring in that dependency explicitly (pull request at [2]) just to allow this substitution to work correctly and then rebuilt the quickstart project with this newer `quarkus-hibernate-orm`. This time there were no more errors with:

```
<additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
```

But of course, that doesn't mean we won't run into exceptions at runtime on first access. So I decided to remove that param and rebuild and this time I see lesser number of errors - 3 of them to be precise. All of those 3 are:

```
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getExecutableTypeAnnotationBytes(Executable) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
```    

```
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getRawTypeAnnotations() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
```

```    
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
```

So it appears that the `java.lang.Class` is actually substituted (right in the substrate VM here[3]) and that substituted version doesn't support these `getExecutableTypeAnnotationBytes` `getRawTypeAnnotations` and `getConstantPool` methods. But some piece of code within this quarkus application seems to be ending up calling it. I have the analysis tree generated out of this run but I'm yet to fully understand how to parse these substrate VM logs.

[1] https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/hibernate-core.gradle#L52
[2] https://github.com/quarkusio/quarkus/pull/4440
[3] https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/hub/DynamicHub.java#L97

The big question is: why does it work when you really use Hibernate ORM :).

On Tue, Oct 8, 2019 at 4:08 PM Jaikiran <notifications@github.com> wrote:

> So debugging this a bit, I have been able to narrow this down and solve
> one part of this issue. I can reproduce the issue consistently if I use a
> simple application with a dependency on the quarkus-hibernate-orm
> extension (and no other extension) and then run:
>
> mvn clean verify -Pnative
>
> (even against latest upstream master). With that command it throws a bunch
> of substrate exceptions which are hard to understand. So I introduced the
>
> <additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
>
> in my pom.xml for the native image build and that showed up some
> interesting exceptions. For example of them is:
>
> Fatal error: java.lang.NoClassDefFoundError
> 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
> 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
> 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
> 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
> 	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
> 	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
> 	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:461)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:310)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:448)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:113)
> Caused by: java.lang.NoClassDefFoundError: javax/security/jacc/PolicyContextException
> 	at java.lang.Class.getDeclaredMethods0(Native Method)
> 	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
> 	at java.lang.Class.getDeclaredMethods(Class.java:1975)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleDeletedClass(AnnotationSubstitutionProcessor.java:437)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:270)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:230)
> 	at com.oracle.svm.hosted.NativeImageGenerator.createDeclarativeSubstitutionProcessor(NativeImageGenerator.java:875)
> 	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:824)
> 	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:524)
> 	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:444)
> 	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
> 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
> 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
> 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
> 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
> Caused by: java.lang.ClassNotFoundException: javax.security.jacc.PolicyContextException
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> 	... 15 more
>
>
> Debugging this showed that, this exception happens because the Hibernate
> ORM extension has this @Delete substitution:
>
> @TargetClass(className = "org.hibernate.secure.internal.StandardJaccServiceImpl")
> @Delete
> public final class Delete_StandardJaccServiceImpl {
> }
>
> (I'm not an exepert, but) the way Substrate seems to deal with deletes is
> that it loads the target class (
> org.hibernate.secure.internal.StandardJaccServiceImpl) first and then
> does a getDeclaredMethods and getDeclaredFields on it, which obviously
> means that it needs the necessary dependency classes to be present (or else
> you will run into the exception above). StandardJaccServiceImpl has
> depedency on jacc classes. Looking at the quarkus-hiberate-orm extension
> it doesn't list the dependency on the artifact which brings in these
> javax.security.jacc.* classes. Looking at the upstream hibernate-orm
> project, that dependency is actually a provided scope dependency[1], so
> the inclusion of hibernate-core artifact doesn't bring this in. So I
> updated our quarkus-hibernate-orm extension to bring in that dependency
> explicitly (pull request at [2]) just to allow this substitution to work
> correctly and then rebuilt the quickstart project with this newer
> quarkus-hibernate-orm. This time there were no more errors with:
>
> <additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
>
> But of course, that doesn't mean we won't run into exceptions at runtime
> on first access. So I decided to remove that param and rebuild and this
> time I see lesser number of errors - 3 of them to be precise. All of those
> 3 are:
>
> Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getExecutableTypeAnnotationBytes(Executable) is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
>
> Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getRawTypeAnnotations() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
>
> Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
>
> So it appears that the java.lang.Class is actually substituted (right in
> the substrate VM here[3]) and that substituted version doesn't support
> these getExecutableTypeAnnotationBytes getRawTypeAnnotations and
> getConstantPool methods. But some piece of code within this quarkus
> application seems to be ending up calling it. I have the analysis tree
> generated out of this run but I'm yet to fully understand how to parse
> these substrate VM logs.
>
> [1]
> https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/hibernate-core.gradle#L52
> [2] #4440 <https://github.com/quarkusio/quarkus/pull/4440>
> [3]
> https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/hub/DynamicHub.java#L97
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkus/issues/2482?email_source=notifications&email_token=AAJYOBL2J3C43QKXCRK66Y3QNSH4LA5CNFSM4HNS5AQKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAUJOKA#issuecomment-539531048>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAJYOBMNF4MNI47PACJR2V3QNSH4LANCNFSM4HNS5AQA>
> .
>

> OK but we have projects working? Typically, this very simple project work OK: https://github.com/gsmet/hibernate-search-demo .

> So there's something that makes it work somehow.

Actually, even that project fails *if I add the following in the pom.xml* and run `mvn clean verify -Pnative`:
```
@@ -106,6 +106,8 @@
                 </goals>
                 <configuration>
                   <enableHttpUrlHandler>true</enableHttpUrlHandler>
+                  <additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
+
                 </configuration>
```
It fails with:
```
[INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] graalvm/graalvm-ce-19.2.0.1/Contents/Home/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dio.netty.leakDetection.level=DISABLED -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dio.netty.allocator.maxOrder=1 -J-Dvertx.disableDnsResolver=true --report-unsupported-elements-at-runtime --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar hibernate-search-demo-1.0-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+PrintAnalysisCallTree -H:Log=registerResource: -H:-AddAllCharsets -H:EnableURLProtocols=http -H:-JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace
[hibernate-search-demo-1.0-SNAPSHOT-runner:94789]    classlist:  10,352.37 ms
[hibernate-search-demo-1.0-SNAPSHOT-runner:94789]        setup:     400.41 ms
Fatal error: java.lang.NoClassDefFoundError
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:461)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:310)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:448)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:113)
Caused by: java.lang.NoClassDefFoundError: javax/security/jacc/PolicyContextException
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.getDeclaredMethods(Class.java:1975)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleDeletedClass(AnnotationSubstitutionProcessor.java:437)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:270)
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:230)
	at com.oracle.svm.hosted.NativeImageGenerator.createDeclarativeSubstitutionProcessor(NativeImageGenerator.java:875)
	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:824)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:524)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:444)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: javax.security.jacc.PolicyContextException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 15 more
```
Quarkus version : latest upstream master
Graal VM version: 19.2.0.1

Removing that ```--report-unsupported-elements-at-runtime``` gets it to pass fine.

None of this looks right, so let me take a step back and see what part of this is Graal specific and what part is a genuine error that we need to sort out in Quarkus extension.

I've had numerous problems with "report-unsupported-elements-at-runtime";
if you're using it the compiler follows quite different logic in the flow
analysis.

I don't think we should support that mode, we certainly don't test for it
so please don't use it :)

On Tue, 8 Oct 2019 at 16:08, Jaikiran <notifications@github.com> wrote:

> OK but we have projects working? Typically, this very simple project work
> OK: https://github.com/gsmet/hibernate-search-demo .
>
> So there's something that makes it work somehow.
>
> Actually, even that project fails *if I add the following in the pom.xml*
> and run mvn clean verify -Pnative:
>
> @@ -106,6 +106,8 @@
>                  </goals>
>                  <configuration>
>                    <enableHttpUrlHandler>true</enableHttpUrlHandler>
> +                  <additionalBuildArgs>--report-unsupported-elements-at-runtime</additionalBuildArgs>
> +
>                  </configuration>
>
> It fails with:
>
> [INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] graalvm/graalvm-ce-19.2.0.1/Contents/Home/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dio.netty.leakDetection.level=DISABLED -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dio.netty.allocator.maxOrder=1 -J-Dvertx.disableDnsResolver=true --report-unsupported-elements-at-runtime --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar hibernate-search-demo-1.0-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:+PrintAnalysisCallTree -H:Log=registerResource: -H:-AddAllCharsets -H:EnableURLProtocols=http -H:-JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace
> [hibernate-search-demo-1.0-SNAPSHOT-runner:94789]    classlist:  10,352.37 ms
> [hibernate-search-demo-1.0-SNAPSHOT-runner:94789]        setup:     400.41 ms
> Fatal error: java.lang.NoClassDefFoundError
> 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
> 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
> 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
> 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
> 	at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:598)
> 	at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
> 	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:461)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:310)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:448)
> 	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:113)
> Caused by: java.lang.NoClassDefFoundError: javax/security/jacc/PolicyContextException
> 	at java.lang.Class.getDeclaredMethods0(Native Method)
> 	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
> 	at java.lang.Class.getDeclaredMethods(Class.java:1975)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleDeletedClass(AnnotationSubstitutionProcessor.java:437)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:270)
> 	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:230)
> 	at com.oracle.svm.hosted.NativeImageGenerator.createDeclarativeSubstitutionProcessor(NativeImageGenerator.java:875)
> 	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:824)
> 	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:524)
> 	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:444)
> 	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
> 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
> 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
> 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
> 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
> Caused by: java.lang.ClassNotFoundException: javax.security.jacc.PolicyContextException
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> 	... 15 more
>
> Quarkus version : latest upstream master
> Graal VM version: 19.2.0.1
>
> Removing that --report-unsupported-elements-at-runtime gets it to pass
> fine.
>
> None of this looks right, so let me take a step back and see what part of
> this is Graal specific and what part is a genuine error that we need to
> sort out in Quarkus extension.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkus/issues/2482?email_source=notifications&email_token=AAAKMTKOT6ON3FCFIGRWNLTQNSO5HA5CNFSM4HNS5AQKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAUQHJY#issuecomment-539558823>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAKMTJRXLQBAASF6VKSYNLQNSO5HANCNFSM4HNS5AQA>
> .
>

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -107,6 +107,16 @@ public final class HibernateOrmProcessor {
      */
     HibernateOrmConfig hibernateConfig;
 
+    // We do our own enhancement during the compilation phase, so disable any
+    // automatic entity enhancement by Hibernate ORM
+    // This has to happen before Hibernate ORM classes are initialized: see
+    // org.hibernate.cfg.Environment#BYTECODE_PROVIDER_INSTANCE
+    @BuildStep
+    public SystemPropertyBuildItem enforceDisableRuntimeEnhancer() {
+        return new SystemPropertyBuildItem(AvailableSettings.BYTECODE_PROVIDER,
+                org.hibernate.cfg.Environment.BYTECODE_PROVIDER_NAME_NONE);
+    }
+
     @BuildStep
     List<HotDeploymentWatchedFileBuildItem> hotDeploymentWatchedFiles(LaunchModeBuildItem launchMode) {
         List<HotDeploymentWatchedFileBuildItem> watchedFiles = new ArrayList<>();

--- a/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
+++ b/extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java
@@ -170,12 +170,13 @@ public final class HibernateOrmProcessor {
         // remember how to run the enhancers later
         domainObjectsProducer.produce(domainObjects);
 
-        if (!hasEntities(domainObjects, nonJpaModelBuildItems)) {
+        final boolean enableORM = hasEntities(domainObjects, nonJpaModelBuildItems);
+        recorder.callHibernateFeatureInit(enableORM);
+
+        if (!enableORM) {
             // we can bail out early
             return;
         }
-
-        recorder.callHibernateFeatureInit();
 
         // handle the implicit persistence unit
         List<ParsedPersistenceXmlDescriptor> allDescriptors = new ArrayList<>(explicitDescriptors.size() + 1);

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/Hibernate.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/Hibernate.java
@@ -1,6 +1,5 @@
 package io.quarkus.hibernate.orm.runtime;
 
-import org.hibernate.cfg.AvailableSettings;
 import org.jboss.logging.Logger;
 
 public class Hibernate {
@@ -9,13 +8,6 @@ public class Hibernate {
         // Override the JPA persistence unit resolver so to use our custom boot
         // strategy:
         PersistenceProviderSetup.registerPersistenceProvider();
-
-        // We do our own enhancement during the compilation phase, so disable any
-        // automatic entity enhancement by Hibernate ORM
-        // This has to happen before Hibernate ORM classes are initialized: see
-        // org.hibernate.cfg.Environment#BYTECODE_PROVIDER_INSTANCE
-        System.setProperty(AvailableSettings.BYTECODE_PROVIDER,
-                org.hibernate.cfg.Environment.BYTECODE_PROVIDER_NAME_NONE);
     }
 
     public static void featureInit(boolean enabled) {

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/Hibernate.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/Hibernate.java
@@ -18,8 +18,10 @@ public class Hibernate {
                 org.hibernate.cfg.Environment.BYTECODE_PROVIDER_NAME_NONE);
     }
 
-    public static void featureInit() {
-        Logger.getLogger("org.hibernate.quarkus.feature").debug("Hibernate Features Enabled");
+    public static void featureInit(boolean enabled) {
+        if (enabled) {
+            Logger.getLogger("org.hibernate.quarkus.feature").debug("Hibernate Features Enabled");
+        }
     }
 
 }

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java
@@ -30,8 +30,13 @@ public class HibernateOrmRecorder {
         Logger.getLogger("io.quarkus.hibernate.orm").debugf("List of entities found by Quarkus deployment:%n%s", entities);
     }
 
-    public void callHibernateFeatureInit() {
-        Hibernate.featureInit();
+    /**
+     * The feature needs to be initialized, even if it's not enabled.
+     *
+     * @param enabled Set to false if it's not being enabled, to log appropriately.
+     */
+    public void callHibernateFeatureInit(boolean enabled) {
+        Hibernate.featureInit(enabled);
     }
 
     public BeanContainerListener initializeJpa(boolean jtaEnabled) {

--- a/integration-tests/jpa-without-entity/pom.xml
+++ b/integration-tests/jpa-without-entity/pom.xml
@@ -1,0 +1,125 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-integration-test-jpa-without-entity</artifactId>
+
+    <name>Quarkus - Integration Tests - JPA without entities</name>
+    <description>Verifies against odd problems which manifest when there are no entities</description>
+    <dependencies>
+        <!-- Include the JPA extensions
+            but we'll intentionally not use it:
+            verify that this is ok -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-hibernate-orm</artifactId>
+        </dependency>
+
+        <!-- All other test dependencies -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jdbc-postgresql</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-hibernate-orm</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-jsonb</artifactId>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <profiles>
+        <profile>
+            <id>native-image</id>
+            <activation>
+                <property>
+                    <name>native</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-failsafe-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                    <goal>verify</goal>
+                                </goals>
+                                <configuration>
+                                    <systemProperties>
+                                        <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
+                                    </systemProperties>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>io.quarkus</groupId>
+                        <artifactId>quarkus-maven-plugin</artifactId>
+                        <version>${project.version}</version>
+                        <executions>
+                            <execution>
+                                <id>native-image</id>
+                                <goals>
+                                    <goal>native-image</goal>
+                                </goals>
+                                <configuration>
+                                    <reportErrorsAtRuntime>false</reportErrorsAtRuntime>
+                                    <cleanupServer>true</cleanupServer>
+                                    <enableHttpUrlHandler>true</enableHttpUrlHandler>
+                                    <graalvmHome>${graalvmHome}</graalvmHome>
+                                    <enableJni>false</enableJni>
+                                    <debugBuildProcess>false</debugBuildProcess>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+    </profiles>
+
+
+</project>

--- a/integration-tests/jpa-without-entity/src/main/java/org/test/HelloResource.java
+++ b/integration-tests/jpa-without-entity/src/main/java/org/test/HelloResource.java
@@ -1,0 +1,16 @@
+package org.test;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/y")
+public class HelloResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+}

--- a/integration-tests/jpa-without-entity/src/test/java/org/test/HelloResourceTest.java
+++ b/integration-tests/jpa-without-entity/src/test/java/org/test/HelloResourceTest.java
@@ -1,0 +1,22 @@
+package org.test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class HelloResourceTest {
+
+    @Test
+    public void testHelloEndpoint() {
+        given()
+                .when().get("/y")
+                .then()
+                .statusCode(200)
+                .body(is("hello"));
+    }
+
+}

--- a/integration-tests/jpa-without-entity/src/test/java/org/test/NativeHelloResourceIT.java
+++ b/integration-tests/jpa-without-entity/src/test/java/org/test/NativeHelloResourceIT.java
@@ -1,0 +1,9 @@
+package org.test;
+
+import io.quarkus.test.junit.NativeImageTest;
+
+@NativeImageTest
+public class NativeHelloResourceIT extends HelloResourceTest {
+
+    // Execute the same tests but in native mode.
+}

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -85,6 +85,7 @@
         <module>narayana-jta</module>
         <module>elytron-security-jdbc</module>
         <module>vertx-graphql</module>
+        <module>jpa-without-entity</module>
     </modules>
 
     <build>

